### ISSUE TITLE
[Feature Request] Accept host names as a target in addition to ip addresses

### ISSUE THREAD

**Issue by aiotter** [hronro/iPortForwarder#5]
I really like this app and using it in my daily work. Thanks for your great work!

I'm using [OrbStack](https://orbstack.dev) and it exposes Kubernetes resources with their service names.
It means you can access your ClusterIP by `service-name.default.svc.cluster.local:80` or so.
It would be really great if iPortForwarder can forward `localhost:8080` to `service-name.default.svc.cluster.local:80`.

**Comment by hronro** [comment]
Good idea, but I'm currently short on time to pursue it. Don't anticipate a quick implementation.

**Comment by hronro** [comment]
Now, when iPortForwarder detects that you have not entered a valid IPv4 or IPv6 address, it will perform a DNS lookup to obtain the real IP address. This means you can forward not only addresses like `service-name.default.svc.cluster.local`, but also public domains such as `www.google.com`. Additionally, I have added an icon next to the input box to indicate whether the app will treat the entry as an IP address or a domain.

When the DNS lookup fails, you will receive an alert.

A quick demo here:

https://github.com/user-attachments/assets/45f5a1fa-c3e5-4a97-be9a-e52d852eaa03

**Comment by aiotter** [comment]
Wow, it looks really nice! I'm looking forward to trying it on a new release.

**Comment by hronro** [comment]
@aiotter You can try it now by downloading the latest build in CI: https://github.com/hronro/iPortForwarder/actions/runs/12880822694

**Comment by aiotter** [comment]
I tried it and it's working nicely!

The icon for domain mode looks weird a bit for me, because "Do" is not often used as an abbreviation for "domain".
It is obvious whether the one inputs a domain name or an ip address. I don't think the icon is very helpful.

Anyway, now it got far easier to use with domain names. I really appreciate it!

**Comment by hronro** [comment]
@aiotter Do you think we should remove the icon, or do you have a better solution?

**Comment by aiotter** [comment]
@hronro I have no strong opinion about it, but you can remove the icon in my idea. I think it will cause less confusion without the icon.

**Comment by hronro** [comment]
To reduce confusion, I decided to only display the icon for valid IP addresses. No icon will appear if the input is empty or a domain name.

**Comment by hronro** [comment]
This feature is now included in the release: https://github.com/hronro/iPortForwarder/releases/tag/v1.0.0
